packages/@react-aria/menu/src/useSubmenuTrigger.ts

```diff
@@ -235,8 +259,15 @@ export function useSubmenuTrigger<T>(props: AriaSubmenuTriggerProps, state: Subm
     },
     submenuProps,
     popoverProps: {
+      // TODO: this is a bit problematic since nested subdialogs won't dismiss when clicking into a parent subdialog since this makes them non-dismissible
+      // Normally shouldCloseonInteractOutside above should be sufficient but all the subdialogs are also react-aria-top-layer and thus isElementInChildScope is
+      // always true and thus useOverlay's useInteractOutside logic will early return
+      // This also cause the subdialog to automatically close because the newly opened subdialog might cause a scroll due to VO focusing the input field in the subdialog
+      // Perhaps I could make subdialogs not have this prop but then screen readers wouldn't be able to navigate to the trigger and user won't be able to hover
+      // the other items in the parent menu when the subdialog is open.
       isNonModal: true,
```

Comment:
Haven't quite figured this one out, but the basic case is opening multiple subdialogs and then clicking into the one of the subdialog's parent subdialog. Maybe we can special case this? Introduce something other than `isNonModal` that is a bit more flexible?

Reply:
- should keep isNonModal behavior for interaction with elements outside the modal
- should not keep isNonModal behavior for isDismissible (needs to be dismissible so you can click)/shouldCloseOnInteractionOutside etc